app/models/katello/concerns/audit_associations_extensions.rb

```diff
@@ -0,0 +1,20 @@
+module Katello
+  module Concerns
+    module AuditAssociationsExtensions
+      def self.included(base)
+        base.include InstanceMethods
+      end
+      module InstanceMethods
+        include ::AuditAssociations::InstanceMethods
+
+        private
+
+        def find_association_class(name)
+          class_name = self.class.reflect_on_association(name).class_name
+          class_name = "Katello::#{class_name}" unless Object.const_defined?(class_name)
```

Comment:
When is this workaround actually needed? It looks like it is calculated correctly already with the existing code:
```ruby
Host::Managed.reflect_on_association(:content_view).class_name
=> "Katello::ContentView"

Reply:
Here it is displaying correctly because it is mentioned explicitly on [line](https://github.com/Katello/katello/blob/master/app/models/katello/host/content_facet.rb#L8).

Wherever `class_name` mentioned explicitly while doing associations, it is calculating correct value based on `class_name` provided.
But when you try to run below statement, it will give you output without `Katello` scope as `class_name` is not provided. 

```
Katello::ActivationKey.reflect_on_association(:host_collections).class_name
=> "HostCollection"
```

Let's see second example, `class_name` mentioned while defining association between Host::Managed and Katello::Hostcollection on [line](https://github.com/Katello/katello/blob/master/app/models/katello/concerns/host_managed_extensions.rb#L28). so it returns output correctly.

```
Host::Managed.reflect_on_association(:host_collections).class_name
=> "::Katello::HostCollection"
```

That's why overridden this method here.

Reply:
Would it make sense than to instead set the `class_name` for the associations that don't set it correctly? I think that might be simpler and possibly be helpful in other cases.

Reply:
Yes, right!! Agreed on it as we might need it in other cases as well.

May I go and create a separate issue for this? I don't want to mix this with audit things.

Please let me know your opinion.

Reply:
I'm OK with it either way. From a quick check it seems like most of the associations already set the class name, so it shouldn't be too much of an effort. In any case, this will solve the issue of having to overload this method here.

Reply:
After this change, one has to mention `class_name` for each new association in Katello.
Same thing will be applicable to all other plugins as well if there is a scope like Katello.

Reply:
that should be the case anyways, `class_name` should be set whenever the model name cannot be inferred from the association name. AFAIK other plugins don't namespace their models, and this workaround wouldn't work for them anyway.